### PR TITLE
glycin 1.1.1 (new formula)

### PR DESCRIPTION
**Pull request by p-linnane**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
Need some help writing a test, but this should otherwise be good to go.

### CODE REVIEW

File: Formula/g/glycin.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  depends_on "vala" => :build # for vapigen
+  depends_on "cairo"
+  depends_on "fontconfig"
+  depends_on "glib"
+  depends_on "gtk4"
+  depends_on "jpeg-xl"
+  depends_on "libheif"
+  depends_on "librsvg"
+  depends_on "libseccomp"
+  depends_on :linux
```
Comment (carlocab) [inline Formula/g/glycin.rb:32]: No macOS builds? Sad.

Comment (p-linnane) [reply Formula/g/glycin.rb:32]: In upstream's README they state:
>Glycin is based on technologies like memfds, unix sockets, and linux namespaces. It currently only works on Linux. An adoption to other unixoid systems could be possible without usage of the sandbox mechanism. Windows support is currently not planned and might not be feasible.

So maybe at some point?

File: Formula/g/glycin.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  depends_on "little-cms2"
+
+  def install
+    system "meson", "setup", "build", *std_meson_args
+    system "meson", "compile", "-C", "build", "--verbose"
+    system "meson", "install", "-C", "build"
+  end
+
+  test do
+    system "false"
```
Comment (carlocab) [inline Formula/g/glycin.rb:42]:
```suggestion
    (testpath/"glycin-test.c").write <<~C
      #include "glycin-1/glycin.h"
      int main() {
        GlyLoader *loader = gly_loader_new("#{test_fixtures("test.png")}");
        return loader == 0;
      }
    C
    ENV.append "LDLIBS", "-lglycin-1"
    system "make", "glycin-test"
    system "./glycin-test"
```
Might work I think? Haven't tested.

**Review by carlocab — APPROVED** [review]

File: Formula/g/glycin.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  test do
+    (testpath/"glycin-test.c").write <<~C
+      #include "glycin-1/glycin.h"
+      int main() {
+        GlyLoader *loader = gly_loader_new("#{test_fixtures("test.png")}");
+        return loader == 0;
+      }
+    C
+    ENV.append "LDLIBS", "-lglycin-1"
```
Comment (carlocab) [inline Formula/g/glycin.rb:49]:
```suggestion
    ENV.append_to_cflags shell_output("pkg-config --cflags glycin-1").strip
    ENV.append "LDFLAGS", shell_output("pkg-config --libs-only-L").strip
    ENV.append "LDLIBS", shell_output("pkg-config --libs-only-l").strip
```

File: Formula/g/glycin.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    url :stable
+    regex(/glycin[._-]v?(\d+(?:\.\d+)+)\.t/i)
+  end
+
+  depends_on "gettext" => :build # for msgfmt
+  depends_on "gobject-introspection" => :build
+  depends_on "gperf" => :build
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
```
Comment (carlocab) [inline Formula/g/glycin.rb:21]:
```suggestion
  depends_on "pkg-config" => [:build, :test]
```

**Review by carlocab — COMMENTED** [review]
This should help your test failure.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**